support/fc_sort.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+"""Sort file context definitions
+
+The original setfiles sorting algorithm did not take into
+account regular expression specificity. With the current
+strict and targeted policies this is not an issue because
+the file contexts are partially hand sorted and concatenated
+in the right order so that the matches are generally correct.
+The way reference policy and loadable policy modules handle
+file contexts makes them come out in an unpredictable order
+and therefore setfiles (or this standalone tool) need to sort
+the regular expressions in a deterministic and stable way.
+"""
+
+import sys
+import argparse
+from pathlib import Path
+import re
+import functools
+
+
+class FileContext():
+    """ Container class for file context defintions
+    """
+
+    def __init__(self, path, file_type, context):
+        """ Constructor
+        """
+        self.path = path
+        self.file_type = file_type
+        self.context = context
+
+        self.compute_diffdata()
+
+    def compute_diffdata(self):
+        """ Compute the interal values needed for comparing two file context definitions
+        """
+
+        self.meta = False
+        self.stem_len = 0
+        self.str_len = 0
+
+        skip_escaped = False
+
+        for char in self.path:
+            if skip_escaped:
+                skip_escaped = False
+                continue
+
+            if char in ('.', '^', '$', '?', '*', '+', '|', '[', '(', '{',):
+                self.meta = True
+            if char == '\\':
+                skip_escaped = True
+
+            if not self.meta:
+                self.stem_len += 1
+
+            self.str_len += 1
+
+
+def compare_fcds(a, b):
```

Comment:
My preference would have this as part of the FileContext class, then the sorted() use would work naturally. I think sorted uses `__lt__` and `__eq__` and maybe `__ne__`.

Reply:
done